build.rs

```diff
@@ -14,7 +14,59 @@
 // You should have received a copy of the GNU General Public License
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
+//! Build script to ensure that PVF worker binaries are always built whenever polkadot is built.
+//!
+//! This is needed because `default-members` does the same thing, but only for `cargo build` -- it
+//! does not work for `cargo run`.
+
+use std::{env::var, path::Path, process::Command};
+
+use polkadot_node_core_pvf::{EXECUTE_BINARY_NAME, PREPARE_BINARY_NAME};
+
+// TODO: unskip
+#[rustfmt::skip]
 fn main() {
+	// Build additional artifacts if a single package build is explicitly requested.
+	{
+		let cargo   = dbg!(var("CARGO").expect("`CARGO` env variable is always set by cargo"));
+		let target  = dbg!(var("TARGET").expect("`TARGET` env variable is always set by cargo"));
+		let profile = dbg!(var("PROFILE").expect("`PROFILE` env variable is always set by cargo"));
+		let out_dir = dbg!(var("OUT_DIR").expect("`OUT_DIR` env variable is always set by cargo"));
+		let target_dir = format!("{}/workers", out_dir);
+
+		// assert!(false);
+
+		// TODO: opt-level, debug, features, etc.
```

Comment:
I added this `build.rs` script to fix the issue where `cargo run` does not build the worker binaries. This is still WIP but it needs to get out soon, ideally before the next release. I didn't open a separate PR but tagged this onto yours because it depends on the workers being their own crates.

But how do we forward everything from the cargo command that was called. For features we will have to iterate over all env vars to get them and pass them on. For things like opt-level, lto etc. (there are more) I don't see how to pass them on, but I *think* the child cargo process inherits it from env vars, but am not sure.

Reply:
I looked into whether profile settings are passed along, and they are! Env vars like `CARGO_CFG_OVERFLOW_CHECKS` (see https://doc.rust-lang.org/cargo/reference/environment-variables.html#configuration-environment-variables) are passed along. So this
script is basically good-to-go, modulo a couple of small TODOs.

Reply:
I should undo the change removing `default-members`. I thought it wouldn't be necessary anymore with the build script, but I realized that `cargo install` will not work for the worker binaries without it.

Reply:
I wonder if we can experiment with moving the build script to `cli` crate or something, that way it will also work for malus which depends on that crate. Might make things cleaner. But since we should get this PR in ASAP (hard deadline by Aug 18 before the switch to monorepo), it can be a followup.